### PR TITLE
feat: Handle standard error for API requests

### PR DESCRIPTION
**Pull request by abhiyaantrix**
- Setup common error handler and responder
- Rescue from few common errors
- Add catch all route to handle missing endpoints
- Create errors in I18n
- Write specs
- Configure some missing layout cops for Rubocop

## Summary by Sourcery

Implement a standardized error handling mechanism for API requests by introducing a common error handler module. Enhance route configuration to include a catch-all route for missing endpoints and update I18n for error messages. Improve code quality by configuring additional Rubocop layout cops and expand test coverage to validate error handling.

New Features:
- Introduce a common error handler module to manage standard errors for API requests, providing consistent error responses.

Enhancements:
- Add a catch-all route in the API namespace to handle missing endpoints, improving error handling for undefined routes.
- Enhance the I18n configuration to include error messages for various HTTP status codes, ensuring better localization support.

Build:
- Configure additional layout cops in Rubocop to enforce consistent code formatting, specifically for parameter and argument alignment.

Tests:
- Add new test cases to verify the handling of various errors in the API, ensuring that the correct HTTP status codes and messages are returned.

### CODE REVIEW

**Comment by abhiyaantrix** [comment]
@sourcery-ai review

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,4 +1,9 @@
 # frozen_string_literal: true
 
 class ApplicationController < ActionController::API
+
+  def route_not_found
```
[the remark this answers is not in the document]
Comment (abhiyaantrix) [reply app/controllers/application_controller.rb]: Not sure, `ErrorHandler` is only concerned about capturing and handling the error.
There is no action there and no raising any errors.

[the remark this answers is not in the document]
Comment (arnabry11) [reply app/controllers/application_controller.rb]: Plus `ActionController::RoutingError` is the default error raised by rails when route is not found. It should get rescued automatically via your ErrorHandler. If its not happening, then keep `def route_not_found` but I agree it should be moved to ErrorHandler concern

**Comment by arnabry11** [comment]
BTW, sourcery-ai is giving some good insights

File: app/controllers/application_controller.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (abhiyaantrix) [reply app/controllers/application_controller.rb]: 1. `route_not_found` is being called in routes.rb explicitly
2. It is also not only limited to `API::V1` however it is responding the Error using method from the `ErrorHandler` so it does indeed depend on `ErrorHandler`

My concern is that if we inadvertently change something with `ErrorHandler` later the catch all route might break.
Perhaps some tests on `ApplicationController` would be helpful.

[the remark this answers is not in the document]
Comment (abhiyaantrix) [reply app/controllers/application_controller.rb]: Made the change, and for now moved the catch-all route to `API::V1` namespace only. This leaves us scope to apply different formatting strategies for `API::V2` if needed.

**Comment by abhiyaantrix** [comment]
@sourcery-ai review

**Pull request merged**